sourced/ml/mining/utils/fs.py

```diff
@@ -2,6 +2,18 @@
 from pathlib import Path
 
 
+class IsNotAFileError(Exception):
```

Comment:
You should derive these three exceptions from OSError, similar to https://docs.python.org/3/library/exceptions.html#FileExistsError

Reply:
Nice, done